### PR TITLE
[mesheryctl] Backup meshconfig: Version fixes

### PR DESCRIPTION
**Pull request by metonymic-smokey**
Signed-off-by: metonymic-smokey <ahuja.aditi@gmail.com>

**Description**
This PR backs up invalid configs in `version`, fixes the panics when an invalid config is found and has minor grammatical errors fixed.  

image

image

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: mesheryctl/internal/cli/root/version.go [hunk 1]
```diff
@@ -130,7 +146,7 @@ func checkMesheryctlClientVersion(build string) {
 	}
 	// If user is running an outdated release, let them know.
 	if res.Outdated {
-		logrus.Info("\n  ", build, " is not the latest release. Update to v", res.Current, ".")
+		logrus.Info("\n  ", build, " is not the latest release. Update to ", res.Current, ".")
```
Comment (piyushsingariya) [inline mesheryctl/internal/cli/root/version.go:149]: @metonymic-smokey  mesheryctl was printing `vv0.x.x` before?

Comment (metonymic-smokey) [reply mesheryctl/internal/cli/root/version.go:149]: Yep @piyushsingariya 
image

Comment (metonymic-smokey) [reply mesheryctl/internal/cli/root/version.go:149]: Just realised this today

**Review by piyushsingariya — APPROVED** [review]
loved the concept behind this, and this looks great!

File: mesheryctl/internal/cli/root/version.go [hunk 2]
```diff
@@ -47,21 +47,39 @@ var versionCmd = &cobra.Command{
...
-					return nil
+					logrus.Errorf("unable to create config file")
 				}
-				fmt.Println("current meshconfig backed up to ~/.meshery/config.bak and new meshconfig generated")
-				err = utils.CreateConfigFile()
+
+				// Add Token to context file
+				err = config.AddTokenToConfig(utils.TemplateToken, utils.DefaultConfigPath)
+				if err != nil {
+					logrus.Errorf("unable to add token to config")
```
Comment (leecalcote) [inline mesheryctl/internal/cli/root/version.go:62]: Let's switch to the MeshKit-style errors.

**Review by navendu-pottekkat — COMMENTED** [review]
The reason we have to check for invalid config here is because this is a root command right? All the other sub commands handles this through root right? @metonymic-smokey

This looks good if this is the case, we can merge after you switch to Meshkit errors.

**Comment by metonymic-smokey** [comment]
> The reason we have to check for invalid config here is because this is a root command right? All the other sub commands handles this through root right? @metonymic-smokey
> 
> This looks good if this is the case, we can merge after you switch to Meshkit errors.

Yes @navendu-pottekkat 

I think switching logrus to meshkit errors will take some time since that has a dependency to be resolved, in the form of a singleton file to create handlers.  
Since this PR fixes some bugs/panics, I'd suggest this is merged if everything other than the errors is okay.  
Once the singleton file is in place, I can migrate `version.go` to use meshkit style errors. 

Thanks!

**Review by navendu-pottekkat — APPROVED** [review]
Looks good @metonymic-smokey

**Pull request merged**